src/sfdt/__tests__/populate.test.js

```diff
@@ -113,28 +115,38 @@ describe('Populate', () => {
 		};
 		const originalInlines = getInline(sfdt);
 
-		const updatedSfdt = populate(data, { ...sfdt });
+		const updatedSfdt = populate(data, {...sfdt});
 
 		const currentInlines = getInline(updatedSfdt);
 
-		//we should have `Line 1` injected in a present inline, line 2 and 3 are new inlines 
+		//we should have `Line 1` injected in a present inline, line 2 and 3 are new inlines
 		// and 2 inlines with vertival tab as separator => size + 4
 		expect(currentInlines.length).toEqual(originalInlines.length + 4);
 
 		const targetInlines = [
-			{ text: 'starting' },
-			{ bookmarkType: 0, name: 'DATA::K1' },
-			{ text: 'Line 1' },
-			{ text: '\u000b' },
-			{ text: 'Line 2' },
-			{ text: '\u000b' },
-			{ text: 'Line 3' },
-			{ bookmarkType: 1, name: 'DATA::K1' },
-			{ bookmarkType: 0, name: 'DATA::K2' },
-			{ text: 'false' },
-			{ bookmarkType: 1, name: 'DATA::K2' },
-			{ text: 'ending' }];
+			{text: 'starting'},
+			{bookmarkType: 0, name: 'DATA::K1'},
+			{text: 'Line 1'},
+			{text: '\u000b'},
+			{text: 'Line 2'},
+			{text: '\u000b'},
+			{text: 'Line 3'},
+			{bookmarkType: 1, name: 'DATA::K1'},
+			{bookmarkType: 0, name: 'DATA::K2'},
+			{text: 'false'},
+			{bookmarkType: 1, name: 'DATA::K2'},
+			{text: 'ending'}
+		];
 
 		expect(currentInlines).toEqual(targetInlines);
 	});
 });
+
+describe('Remove empty section from sfdt', () => {
+	it('Removes whole section if block is empty', () => {
+		const name = 'COND::e2fb3aa8-6f6f-45da-bae5-b0c2d8989686';
+		const updatedSfdt = toggleBookmark(emptyBlockSection, name, false);
+		const finalSfdt = populate({test: 'test'}, updatedSfdt);
+		expect(get(finalSfdt, 'sections').length).toBe(2);
```

Comment:
you should add a check before doing the test to ensure the number of sections before the update is indeed higher than after
In case someone go to change the sfdt itself..

Reply:
you mean i i need to create another test case which confirms the total number of sections to be 3(which is in current sfdt)